src/snowcli/cli/common/sql_execution.py

```diff
@@ -35,6 +35,8 @@ def use_role(self, new_role: str):
         is_different_role = new_role.lower() != prev_role.lower()
         if is_different_role:
             self._execute_query(f"use role {new_role}")
-        yield
-        if is_different_role:
-            self._execute_query(f"use role {prev_role}")
+        try:
```

Comment:
No, it's to catch any exceptions that happen within the `with` clause when this is used as a context manager. So that if your code raises an exception the original role is applied again.